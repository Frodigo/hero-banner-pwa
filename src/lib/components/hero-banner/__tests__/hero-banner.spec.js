import React from 'react';
import renderer from 'react-test-renderer';
import HeroBanner from '../hero-banner';
import HeroBannerAction from '../hero-banner-button';
import HeroBannerFigure from '../hero-banner-figure';
import HeroBannerHeading from '../hero-banner-heading';
import HeroBannerImage from '../hero-banner-image';
import HeroBannerTape from '../hero-banner-tape';

jest.mock('@magento/venia-ui/lib/classify');

describe('Render hero banner', () => {
    const mockedHtml = document.createElement('div');

    test('Display hero banner action button', () => {
        mockedHtml.innerText = 'Shop now';
        mockedHtml.setAttribute('href', '#');

        const tree = renderer
            .create(<HeroBannerAction html={mockedHtml} children={[]} />)
            .toJSON();

        expect(tree).toMatchSnapshot();
        expect(tree.props.href).toEqual('#');
        expect(tree.children[0]).toEqual('Shop now');
    });

    test('Display hero banner heading', () => {
        mockedHtml.innerText = 'Heading';
        const tree = renderer
            .create(<HeroBannerHeading html={mockedHtml} children={[]} />)
            .toJSON();

        expect(tree).toMatchSnapshot();
        expect(tree.children[0]).toEqual('Heading');
    });

    test('Display hero banner on the page', () => {
        const tree = renderer
            .create(<HeroBanner html={mockedHtml} children={[]} />)
            .toJSON();

        expect(tree).toMatchSnapshot();
    });

    test('Display hero banner figure', () => {
        const tree = renderer
            .create(<HeroBannerFigure html={mockedHtml} children={[]} />)
            .toJSON();

        expect(tree).toMatchSnapshot();
    });

    test('Display hero banner image', () => {
        mockedHtml.setAttribute('src','#')
        const tree = renderer
            .create(<HeroBannerImage html={mockedHtml} children={[]} />)
            .toJSON();

        expect(tree).toMatchSnapshot();
    });

    test('Display hero banner tape', () => {
        const tree = renderer
            .create(<HeroBannerTape html={mockedHtml} children={[]} />)
            .toJSON();

        expect(tree).toMatchSnapshot();
    });
});


