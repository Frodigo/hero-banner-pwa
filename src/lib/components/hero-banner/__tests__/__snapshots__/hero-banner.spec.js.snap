// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render hero banner Display hero banner action button 1`] = `
<a
  className="heroBlockAction"
  href="#"
>
  Shop now
</a>
`;

exports[`Render hero banner Display hero banner figure 1`] = `
<figure
  className="heroBlockFigure"
/>
`;

exports[`Render hero banner Display hero banner heading 1`] = `
<h2
  className="heroBlockHeading"
>
  Heading
</h2>
`;

exports[`Render hero banner Display hero banner image 1`] = `
<img
  alt="figure"
  className="heroBlockImage"
  src="#"
/>
`;

exports[`Render hero banner Display hero banner on the page 1`] = `
<div
  className="heroBlock"
/>
`;

exports[`Render hero banner Display hero banner tape 1`] = `
<section
  className="heroBlockTape"
/>
`;
